filebeat/fileset/pipelines.go

```diff
@@ -65,16 +66,17 @@ func (reg *ModuleRegistry) LoadPipelines(esClient PipelineLoader, overwrite bool
 			// check that all the required Ingest Node plugins are available
 			requiredProcessors := fileset.GetRequiredProcessors()
 			reg.log.Debugf("Required processors: %s", requiredProcessors)
-			if len(requiredProcessors) > 0 {
+			// APIs do not exist on serverless
+			if len(requiredProcessors) > 0 && !esClient.IsServerless() {
 				err := checkAvailableProcessors(esClient, requiredProcessors)
 				if err != nil {
-					return fmt.Errorf("error loading pipeline for fileset %s/%s: %v", module.config.Module, fileset.name, err)
+					return fmt.Errorf("error loading pipeline for fileset %s/%s: %w", module.config.Module, fileset.name, err)
 				}
 			}
```

Comment:
Disclaimer: I am not an expert on filebeat so the question may be naibe or obvious :sweat: 

Wouldn't it be a bit cleaner to have 1 empty set of processors for serverless in line 67 and load the regular sets of required processors for regular ElasticSearch and modify the `checkAvailableProcessors` to perform the ES call only if we pass a non-empty set of processors ?

The effect is the same but the code should look a bit cleaner.

Reply:
Maybe? One of the reasons why I chose that was that it made the code fairly explicit, I didn't want to bury the serverless logic too much.